tango/integrations/torch/train.py

```diff
@@ -348,6 +348,14 @@ def _train(
     callbacks: Optional[List[Lazy[TrainCallback]]] = None,
     include_package: Optional[Set[str]] = None,
 ) -> Optional[Model]:
+    # Set random seeds.
+    set_seed_all(config.seed)
+
+    # force torch to be deterministic
+    os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"
+    torch.backends.cudnn.benchmark = False
+    torch.use_deterministic_algorithms(True)
```

Comment:
This has a negative effect on performance, so we probably don't want to do this. If you need this you could always call it from a callback though.

Reply:
Makes sense, I like the callback idea! 
But should the step then still be flagged as `DETERMINISTIC` ? It could be confusing because the training results can vary substantially without these calls.

Reply:
In my opinion, yes, because there are benefits of having the step be marked DETERMINISTIC: caching isn't useful for non DETERMINISTIC steps. Besides, even after calling this torch function you're not guaranteed deterministic results.